protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -303,43 +361,26 @@ fn execute_cheatcode_hint(
             let tx_result = account_tx
                 .execute(blockifier_state, block_context)
                 .expect("Failed to execute declare transaction");
+            // result_segment.
+            let felt_class_hash = felt252_from_hex_string(&class_hash.to_string()).unwrap();
 
-            insert_value_to_cellref!(
-                vm,
-                result,
-                felt252_from_hex_string(&class_hash.to_string()).unwrap()
-            )?;
             // TODO https://github.com/software-mansion/protostar/issues/2024
-            //  in case of errors above, consider not panicking, set an error and return it here
-            //  instead
-            insert_value_to_cellref!(vm, err_code, Felt252::from(0))?;
-            Ok(())
+            //  in case of errors above, consider not panicking,
+            //  set an error and return it here instead
+            insert_at_pointer(vm, &mut result_segment_ptr, Felt252::from(0)).unwrap();
+            insert_at_pointer(vm, &mut result_segment_ptr, felt_class_hash).unwrap();
         }
-        &ProtostarHint::DeclareCairo0 { .. } => todo!(),
-        &ProtostarHint::StartPrank { .. } => todo!(),
-        &ProtostarHint::StopPrank { .. } => todo!(),
-        &ProtostarHint::Invoke { .. } => todo!(),
-        &ProtostarHint::MockCall { .. } => todo!(),
-        ProtostarHint::Deploy {
-            prepared_contract_address,
-            prepared_class_hash,
-            prepared_constructor_calldata_start,
-            prepared_constructor_calldata_end,
-            deployed_contract_address,
-            panic_data_start,
-            panic_data_end,
-        } => {
-            let contract_address = get_val(vm, prepared_contract_address)?;
+        "deploy" => {
+            let contract_address = inputs[0].clone();
             // TODO(#1991) deploy should fail if contract address provided doesn't match calculated
             //  or not accept this address as argument at all.
-            let class_hash = get_val(vm, prepared_class_hash)?;
-            let as_relocatable = |vm, value| {
-                let (base, offset) = extract_buffer(value);
-                get_ptr(vm, base, &offset)
-            };
-            let mut curr = as_relocatable(vm, prepared_constructor_calldata_start)?;
-            let end = as_relocatable(vm, prepared_constructor_calldata_end)?;
-            let calldata = read_data_from_range(vm, curr, end).unwrap();
+            let class_hash = inputs[1].clone();
+
+            let calldata_length: usize = inputs[2].to_usize().unwrap_or(0);
```

Comment:
in what scenario does this `to_usize` fail?

Reply:
In theory `inputs[2]` is of type `Felt252`, therefore it can store more data than `usize`. Practically it should never fail since we append proper length of the array [here](https://github.com/software-mansion-labs/cairo/pull/93/files#diff-954fd47e9dc8a1c411a344ddb12d510aa0b387d21089cae116025ee665cb27faR49-R50)

Reply:
Changed this to just `unwrap`. We shouldn't rely on default values for things like this.